### ISSUE TITLE
A Kind reminder for Decoding method of Qwen models

### ISSUE THREAD

**Issue by 5k5000** [eqimp/hogwild_llm#23]
Dear Authors,

I would like to once again express my deep appreciation for this project.

I have noticed that greedy decoding appears to be used by default in your repository (specifically, new_tokens = logits.argmax(-1)   # <-- greedy generation in the Jupyter examples).

I wanted to kindly note that, according to the [official Qwen documentation](https://qwen.readthedocs.io/en/latest/getting_started/quickstart.html), greedy decoding is not recommended for Qwen3 thinking models, as it may lead to performance degradation and endless repetitions.

<img width="775" height="153" alt="Image" src="https://github.com/user-attachments/assets/84d73023-5f99-4075-a2f2-c8964dedc972" />

I am not sure if this has already come to your attention, so I wanted to share this as a warm reminder.

Thank you again for your excellent work!

**Comment by eqimp** [comment]
Hi, thank you for your kind words!

We use greedy generation only in basic_example.ipynb, which serves as a minimal playground with a simple cache layout. For other notebooks and the experiments in the paper (generation.py), we use:

```
new_tokens = torch.multinomial(logits.softmax(dim=-1), 1).flatten()
    if model.generation_config.do_sample else logits.argmax(-1)
```
It's worth noting that the parallel cache updates and rotation might require different hyperparameters than standard sequential inference. However, we opted for the simplest sampling approach and haven't performed specific hyperparameter tuning for this yet.

Thanks again for your interest in the project!

**Comment by 5k5000** [comment]
Thanks for your clarification
